source/apps/crawl_space/models.py

```diff
@@ -69,23 +70,15 @@ def get_absolute_url(self):
         return reverse('base:project',
             kwargs=dict(project_slug=self.project.slug))
 
-
     def save(self, *args, **kwargs):
-
         if self.pk is None:
-            super(CrawlModel, self).save(*args, **kwargs)
-
+            self.slug = slugify(self.name)
+            # TODO:
+            # Another weird call with a side effect that has to be fixed.
             model_path = self.ensure_model_path()
```

Comment:
I expect that "ensure_model_path" will make sure a directory/path has been created and contains appropriate content.  Is it doing something beyond this?